st2api/st2api/controllers/v1/auth.py

```diff
@@ -97,11 +101,27 @@ def get_all(self, requester_user, show_secrets=None):
         """
         mask_secrets = self._get_mask_secrets(show_secrets=show_secrets,
                                               requester_user=requester_user)
-        api_key_dbs = ApiKey.get_all()
-        api_keys = [ApiKeyAPI.from_model(api_key_db, mask_secrets=mask_secrets)
-                    for api_key_db in api_key_dbs]
 
-        return api_keys
+        if limit and int(limit) > self.max_limit:
```

Comment:
Hm, just curious, why does this need to be duplicated here (we already do it in some base controller)?

Or is it because that base controller only applies and is used with pack resources?